src/TerminalWidget.vala

```diff
@@ -174,6 +137,67 @@ namespace PantheonTerminal {
             this.clickable (regex_strings);
         }
 
+        public void connect_signals () {
+            /* Connect to necessary signals */
+            settings.changed.connect (restore_settings);
+            child_exited.connect (on_child_exited);
+            button_press_event.connect (on_button_press);
+            button_release_event.connect (on_button_release);
+            selection_changed.connect (on_selection_changed);
+        }
+
+        public void disconnect_signals () {
+            settings.changed.disconnect (restore_settings);
+            child_exited.disconnect (on_child_exited);
+            button_press_event.disconnect (on_button_press);
+            button_release_event.disconnect (on_button_release);
+            selection_changed.disconnect (on_selection_changed);
+        }
+
+        private bool on_button_press (Gdk.EventButton event) {
```

Comment:
Now, I don't know if this would work, but I think, this could be done differently by just overriding the default `button_press_event` virtual signal here.